sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py

```diff
@@ -487,3 +490,53 @@ def __init__(self):
         self.signed_service = None
         self.signed_version = None
         self.value = None
+
+class TokenAudience(Protocol):
+    """
+    Protocol that defines what calling functions and attributes should be defined for an Azure Active
+    Directory audience token.
+
+    :param str value: The Azure Active Directory audience to use when forming authorization scopes.
+        For the Language service, this value corresponds to a URL that identifies the Azure cloud
+        where the resource is located. For more information please take a look at:
+        https://learn.microsoft.com/azure/storage/blobs/authorize-access-azure-active-directory
+        Please use one of the static constant members over creating a custom value unless you have
+        a specific scenario for doing so.
+    """
+    _value: str
+
+    def __init__(self, value: str) -> None:
+        self._value = value
+
+    @classmethod
+    def public_audience(cls) -> Self:
+        """Default Audience. Use to acquire a token for authorizing requests to any Azure Storage account.
+        
+        Resource ID: 'https://storage.azure.com/'
+        If no audience is specified, this is the default value.
+
+        :returns: The Default Audience.
+        :rtype: TokenAudience
+        """
+        return cls(STORAGE_OAUTH_SCOPE)
+
+    @classmethod
+    def get_service_account_audience(cls, storageAccountName: str) -> Self:
```

Comment:
Remove this from here, it will be provided in the child classes only and be package specific.